Tools-Override/CodeCoverage.targets

```diff
@@ -0,0 +1,132 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <!--
+    Code coverage package versions go here and in the test-runtime-packages.config
+  -->
+  <PropertyGroup>
+    <OpenCoverVersion>4.6.519</OpenCoverVersion>
+    <ReportGeneratorVersion>2.5.0</ReportGeneratorVersion>
+    <CoverallsUploaderVersion>1.4</CoverallsUploaderVersion>
+  </PropertyGroup>
+  
+  <!-- Coverage options -->
+  <PropertyGroup>
+    <CodeCoverageEnabled>false</CodeCoverageEnabled>
+    <CodeCoverageEnabled Condition="'$(SkipTests)' != 'true' and '$(OS)' == 'Windows_NT' and '$(Coverage)' == 'true' and '$(Performance)' != 'true'">true</CodeCoverageEnabled>
+    <CoverageReportDir Condition="'$(CoverageReportDir)' == ''">$(TestWorkingDir)coverage\</CoverageReportDir>
+
+    <!-- This targets file has two modes one for individual projects and one for all -->
+    <GenerateCodeCoverageReportForAll Condition="'$(GenerateCodeCoverageReportForAll)'==''">false</GenerateCodeCoverageReportForAll>
+    <CoverageEnabledForProject Condition="'$(GenerateCodeCoverageReportForAll)'=='true'">false</CoverageEnabledForProject>
+    <CoverageEnabledForProject Condition="'$(CoverageEnabledForProject)'=='' and '$(IsTestProject)'=='true'">$(CodeCoverageEnabled)</CoverageEnabledForProject>
+
+    <GenerateIndividualCoverageReport Condition="'$(BuildAllProjects)'!='true' and '$(CoverageEnabledForProject)'=='true'">true</GenerateIndividualCoverageReport>
+    <GenerateFullCoverageReport Condition="'$(GenerateCodeCoverageReportForAll)'=='true' and '$(CodeCoverageEnabled)'=='true' and '$(SkipFullCoverageReport)' != 'true'">true</GenerateFullCoverageReport>
+    <GenerateVisitedMethodsReport Condition="'$(GenerateCodeCoverageReportForAll)'=='true' and '$(CodeCoverageEnabled)'=='true' and '$(GenerateVisitedReport)' == 'true'">true</GenerateVisitedMethodsReport>
+
+    <!-- We need to filter the data to only the assembly being tested. Otherwise we will gather tons of data about other assemblies. -->
+    <AssemblyBeingTestedName>$(MSBuildProjectName.Replace('.Tests', '').Replace('.Functional', '').Replace('.Performance', '').Replace('.Unit', ''))</AssemblyBeingTestedName>
```

Comment:
> Is this a problem because of time impact, space impact, something else?

It has a pretty significant impact on disk size, at the very least. Without filtering, I was easily collecting 8+ GB of data + report files. I don't know if that is acceptable, but I could add an option to disable the filtering.